infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogCompactor.java

```diff
@@ -389,7 +343,7 @@ private int writeBatchToCompactionOutput(List<LogData> batch,
 
     private void writeBatchToCompactionOutput(List<LogData> batch,
                                              AbstractLogSegment compactionOutput) {
-        compactionOutput.append(batch);
+        compactionOutput.appendCompacted(batch);
```

Comment:
I think it's cleaner to keep the current method and add a new one, because this new one is only used internally for compactor. Also otherwise a lot of places have to change.